### PR TITLE
fix(a11y): multiple a11y fixes

### PR DESCRIPTION
**Pull request by torgeadelin**
# Description
- SR & Live Region
        - add new componet to handle SR annoucements with delay, as focus imput interrupts the annoucements
        - screen reader announcement no longer announces individual emojis, but rather the count of emojis found OR no emoji found message if none are found
        - add aria-atomic for the aria-live region to ensure the SR reads the whole message

- Navigation
        - disabled top navigation buttons while search input is focused

- Search
        - changed the outer div to a form with role search to improve semantics
        - add aria-hidden on the loupe icon in the search bar
        - add aria-label to the search input

- Search Result & Default Categories
        - render no result message in the preview section, when preview section is not rendered. previously the no result was only shown in the preview, but since the picker already supports hiding it, we don't want to loose the visual cue of "no emojis found"
        - add aria-label to the outer div around the all the moji categories
        - add aria-labelledby for each group of emojis pointing to the section header, and add role = group

### CODE REVIEW

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 1]
```diff
@@ -1067,20 +1096,11 @@ export default class Picker extends Component {
     )
   }
 
+  // This is a hidden live region that announces the number of emojis found only
   renderLiveRegion() {
```
Comment (torgeadelin) [inline packages/emoji-mart/src/components/Picker/Picker.tsx]: I forgot to remove this

Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: removed.

Comment (JanPodmajersky) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: Could you please amend this to previous commit commit? 🙏 So that the history is nicer.

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 2]
```diff
@@ -1096,15 +1097,6 @@ export default class Picker extends Component {
     )
   }
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/Picker.tsx:1103]: This comment removal should be part of first commit 🙏 .

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 3]
```diff
@@ -1082,6 +1082,7 @@ export default class Picker extends Component {
           ref={this.refs.skinToneButton}
           class="skin-tone-button flex flex-auto flex-center flex-middle"
           aria-selected={this.state.showSkins ? 'true' : 'false'}
+          aria-expanded={this.state.showSkins ? 'true' : 'false'}
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/Picker.tsx]: Could you please keep in this commit only this change (the other should be part of the first one 🙏 ) with a brief explanation why is it necessary? Which issue fixes it? Thank you 

Nitpick:
This expresion is repetitive, maybe it might be beneficial to introduce a variable which will keep it
```ts
const ariaSkinToneShown = this.state.showSkins ? 'true' : 'false'
```

File: packages/emoji-mart-data/i18n/en.json [hunk 4]
```diff
@@ -31,6 +27,9 @@
     "6": "Dark"
   },
   "a11y": {
-    "available_emojis": "Available emojis"
+    "available_emojis": "Available emojis",
+    "emojis_found_plural": "emojis found",
```
Comment (JanPodmajersky) [inline packages/emoji-mart-data/i18n/en.json:31]: Thank you for organizing the emojis, what about other languages 🤔 . Do you plan to add them?

File: packages/emoji-mart/src/components/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -111,6 +111,7 @@ export default class Navigation extends PureComponent {
                 type="button"
                 class="flex flex-grow flex-center"
                 role="tab"
+                disabled={this.props.disabled}
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Navigation/Navigation.tsx:114]: Thank you for all the changes, with explanation in commit's body 👏 , but they miss the explanation why is it necessary?

> - Navigation
	- disabled top navigation buttons while search input is focused 

Why is it a problem? Why this solution to disable button is selected? I think disabling buttons should be our last resort: https://adrianroselli.com/2024/02/dont-disable-form-controls.html

After digging deeper I've found that this disabled button really make sense in the current implementation 🙈 as clicking the navigation category does not display it as it is [hidden](https://github.com/slidoapp/emoji-mart/blob/main/packages/emoji-mart/src/components/Picker/Picker.tsx#L942).

Generally we prefer having as little commits only with single issue in it, so that the boundaries of each issue are easily trackable, as now it is hard to follow what is needed for which issue 🤔. Would it make sense for you too + adding explanation why in this case the disabled button makes sense 🙏 ?

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 6]
```diff
@@ -865,17 +881,20 @@ export default class Picker extends Component {
       <div>
         <div class="spacer"></div>
         <div class="flex flex-middle">
-          <div class="search relative flex-grow">
+          <div class="search relative flex-grow" role="presentation">
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/Picker.tsx]: Why presentation role, it should not be necessary since it is used on non semantics `div` element. Or am I missing sth?
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 7]
```diff
@@ -865,17 +881,20 @@ export default class Picker extends Component {
...
               type="search"
               ref={this.refs.searchInput}
               placeholder={I18n.search}
               onClick={this.handleSearchClick}
               onInput={this.handleSearchInput}
               onKeyDown={this.handleEmojisKeyDown}
-              autoComplete="off"
-            ></input>
-            <span class="icon loupe flex">{Icons.search.loupe}</span>
+              aria-label={I18n.search_input_aria_label}
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/Picker.tsx]: Why is this aria-label necessary? Aria-* attributes should be the last resort, but here it seems the semantic input html tag with type search sufffices, doesn't it? In the current version it is announced correctly 🤔 
image

File: packages/emoji-mart/src/components/Picker/Picker.tsx [hunk 8]
```diff
@@ -1202,8 +1225,12 @@ export default class Picker extends Component {
         {this.props.navPosition == 'bottom' && this.renderNav()}
         {this.props.previewPosition == 'bottom' && this.renderPreview()}
         {this.state.showSkins && this.renderSkins()}
-        {this.renderLiveRegion()}
-      </section>
+        <ScreenReaderAnnouncement
+          level="polite"
+          identity="emoji-picker"
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/Picker.tsx]: What is this `identity`?

File: packages/emoji-mart/src/components/Picker/PickerStyles.scss [hunk 9]
```diff
@@ -127,12 +127,15 @@
...
-  width: 1px;
+  border: 0;
+  clip: rect(0, 0, 0, 0);
   height: 1px;
+  margin: -1px;
   overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
```
Comment (JanPodmajersky) [inline packages/emoji-mart/src/components/Picker/PickerStyles.scss:138]: Thank you for fixing this 🙏 .

**Review by JanPodmajersky — CHANGES_REQUESTED** [review]
Thank you for the changes, I am not sure whether all are necessary 🤔 .

File: packages/emoji-mart-data/i18n/en.json (hunk 4 above)
Comment (torgeadelin) [reply packages/emoji-mart-data/i18n/en.json:31]: i wasn't planning to, but I am happy to use google translate and add those as well. Not sure how accurate that will be haha.

File: packages/emoji-mart/src/components/Navigation/Navigation.tsx (hunk 5 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Navigation/Navigation.tsx:114]: Yes, clicking the tabs while the search input is focused, doesn't do anything hence I've disabled them. We found this while testing.

Yeah, sorry about putting everything in one big commit, I was trying to save time.
I'll amend the first commit to include `why` for each change.

File: packages/emoji-mart-data/i18n/en.json (hunk 4 above)
Comment (JanPodmajersky) [reply packages/emoji-mart-data/i18n/en.json:31]: I see, I was only wondering which languages you use, I thought all of them and what would happen if it is not present 🤔 ?

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 6 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: You're not missing anything, I am not sure why I added this. I was probably tried to refactor things and forgot to remove.
Removed.

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 7 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: Yeah, I think it's sufficient by itself, but for non sighted users it would be nice to add more context like "search field to look through emojis", or something like that. It's not clear by default what it's searching for.

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 8 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: artifact from initial implementation, removed

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 3 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: Just for context, this is needed because it triggers an overlay. Adding aria-expanded makes this button a `menu-button`.

Not entirely sure this is 100% correct, as the list with the different skin-tones is not an actual menu.. it's a radiogroup with radio inputs. Having the list with options as radiogroup makes sense, we believe that's correct. But, the a11y team was complaining that the button needs to have role menu-button, which is why we added the aria-expanded. 

I just realised that it also needs the following 2 attributes:
```
aria-controls="skin-tone-options"
aria-haspopup="true"
```

SR now announces:
image

Instead of:
image

Hope this makes sense?

Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: I'm now confident this is fine, as the SR never mentions of a menu.

File: packages/emoji-mart-data/i18n/en.json (hunk 4 above)
Comment (torgeadelin) [reply packages/emoji-mart-data/i18n/en.json:31]: We do have support for all these languages, but AFAIK we have translations for each on our end. (we handle i18n ourselves, and we don't rely on the internal translations).

Comment (torgeadelin) [reply packages/emoji-mart-data/i18n/en.json:31]: I just tested in the demo dev page and it seems that if the translation doesn't exist, it returns equivalent of ''.
I was expecting for english to default, but I think that's another bug.

Comment (JanPodmajersky) [reply packages/emoji-mart-data/i18n/en.json:31]: I see, are you going to take a look at it?

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 7 above)
Comment (JanPodmajersky) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: I see, but the context is already clear for the users, as it is emoji picker, isn't it, I am only thinking about not overwhelming users 🤔 .

**Review by JanPodmajersky — APPROVED** [review]

File: packages/emoji-mart-data/i18n/en.json (hunk 4 above)
Comment (torgeadelin) [reply packages/emoji-mart-data/i18n/en.json:31]: I'm afraid I don't have enough cycles to do that :(, I can raise an issue if that's helpful?

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 7 above)
Comment (torgeadelin) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: For sighted users might be obvious. But it really depends wether the consumer labels this popover content accordingly. I don't think we have a label that says "Popover emoji" anywhere.

File: packages/emoji-mart-data/i18n/en.json (hunk 4 above)
Comment (JanPodmajersky) [reply packages/emoji-mart-data/i18n/en.json:31]: Yeah, it would be much appreciated, thank you.

File: packages/emoji-mart/src/components/Picker/Picker.tsx (hunk 7 above)
Comment (JanPodmajersky) [reply packages/emoji-mart/src/components/Picker/Picker.tsx]: I see, thank you for explanation, it makes sense.
